network/src/commonMain/kotlin/com/wire/kalium/network/NetworkClient.kt

```diff
@@ -30,8 +26,11 @@ import io.ktor.serialization.kotlinx.json.json
  * It's Authenticated, and will use the provided [SessionManager] to fill
  * necessary Authentication headers, and refresh tokens as they expire.
  */
-internal class AuthenticatedNetworkClient(engine: HttpClientEngine, sessionManager: SessionManager) {
-    val httpClient: HttpClient = provideBaseHttpClient(engine, HttpClientOptions.DefaultHost(sessionManager.session().second)) {
+internal class AuthenticatedNetworkClient(
+    engine: HttpClientEngine, sessionManager: SessionManager, serverMetaDataManager: ServerMetaDataManager
+) {
+    val httpClient: HttpClient = provideBaseHttpClient(engine) {
+        installWireDefaultRequest(sessionManager.session().second, serverMetaDataManager)
```

Comment:
Please if it's possible use a datamodel instead of pair to be more clear.

Reply:
If ok, I would like to do this change in a separate PR since it requires quite few changes